spec/nats-tls/nats_tls_config_spec.rb

```diff
@@ -345,7 +352,94 @@ module Bosh::Template::Test
+                },
+                {
+                  'name' => 'healthcheck',
+                  'executable' => '/var/vcap/packages/nats-tls-healthcheck/bin/nats-tls-healthcheck',
+                  'args' => [
+                    '--address',
+                    '10.0.0.1',
+                    '--port',
+                    4222,
+                    '--server-ca',
+                    '/var/vcap/jobs/nats-tls/config/external_tls/ca.pem',
+                    '--server-hostname',
+                    'my-host',
+                    '--client-certificate',
+                    '/var/vcap/jobs/nats-tls/config/client_tls/certificate.pem',
+                    '--client-private-key',
+                    '/var/vcap/jobs/nats-tls/config/client_tls/private_key.pem',
+                    '--user',
+                    'my-user',
+                    '--password',
+                    'my-password'
+                  ]
+                }
+              ]
+            }
+
+            expect(rendered_template).to eq(expected_template)
+          end
+
+          describe 'password authentication is disabled' do
+            before do
+              merged_manifest_properties['nats']['user'] = nil
+              merged_manifest_properties['nats']['password'] = nil
+            end
+
+            it 'renders the template without password authentication properties' do
+              rendered_template = YAML.load(template.render(merged_manifest_properties, consumes: links, spec: spec))
+              expected_template = {
+                'processes' => [
+                  {
+                    'name' => 'nats-tls',
+                    'limits' => {
+                      'open_files' => 100000
+                    },
+                    'executable' => '/var/vcap/packages/gnatsd/bin/gnatsd',
+                    'args' => ['-c', '/var/vcap/jobs/nats-tls/config/nats-tls.conf']
+                  },
+                  {
+                    'name' => 'healthcheck',
+                    'executable' => '/var/vcap/packages/nats-tls-healthcheck/bin/nats-tls-healthcheck',
+                    'args' => [
+                      '--address',
+                      '10.0.0.1',
+                      '--port',
+                      4222,
+                      '--server-ca',
+                      '/var/vcap/jobs/nats-tls/config/external_tls/ca.pem',
+                      '--server-hostname',
+                      'my-host',
+                      '--client-certificate',
+                      '/var/vcap/jobs/nats-tls/config/client_tls/certificate.pem',
+                      '--client-private-key',
+                      '/var/vcap/jobs/nats-tls/config/client_tls/private_key.pem',
+                    ]
+                  }
+                ]
+              }
+
+              expect(rendered_template).to eq(expected_template)
+            end
+          end
+        end
+     end
```

Comment:
@peterellisjones Can you add a few tests here to ensure bosh fails during template renduring with a human-friendly error when the tls certs/keys aren't provided?

Reply:
hi @geofffranks I've added the additional error checking and units tests